src/forms/validation-rules.js

```diff
@@ -221,25 +223,18 @@ export function validCollectedVariables(
  
     return Dictionary.validation_collectedvariable_need_reset;
   }
-
-  if (type === SIMPLE && value[0] || type === TABLE && value[0] ) {
+  
+  if (type === TABLE && value[0] || type === SIMPLE && value[0] ) {
     const typevalue = value[0].type;
     const typeexpectedVariables = expectedVariables[0].type;
     if (
-      value[0] &&
-      value[0].codeListReference &&
-      value[0].codeListReference !== 'undefined'  || 
-      value[0] &&
-      typevalue !== typeexpectedVariables || 
-      value[0] && 
-      typevalue === typeexpectedVariables &&
-      value[0][typevalue] !== expectedVariables[0][typeexpectedVariables]
-   
+      value[0].codeListReference ||
+      typevalue !== typeexpectedVariables ||
+      JSON.stringify(value[0][typevalue]) != JSON.stringify(expectedVariables[0][typeexpectedVariables])
```

Comment:
Are you 100% sure with this check using JSON.stringify ?

Reply:
Oui j'ai les mèmes key value de deux Objects donc j'ai utilisée JSON.stringify pour les  comparer que faire une fonction

Reply:
But are you sure with the order of the properties of each object. 

Because the following statement return false 


```
JSON.stringify({label: 1, value:2}) == JSON.stringify({value:2, label: 1})
```

Reply:
Oui pour l'ordre des properties c'est toujours le même